lib/spack/spack/test/cmd/load.py

```diff
@@ -17,61 +18,94 @@
 install = SpackCommand("install")
 location = SpackCommand("location")
 
-pytestmark = pytest.mark.not_on_windows("does not run on windows")
-
 
+@pytest.mark.parametrize(
+    "shell,set_command,commandsep",
+    (
+        [("--bat", 'set "%s=%s"', "\n")]
+        if sys.platform == "win32"
+        else [("--sh", "export %s=%s", ";\n")]
+    ),
+)
```

Comment:
IMO it's a bit confusing to have a parameter list that always evaluates to 1 list item depending on platform: can you move these variables to the beginning of the function, vs. parameters passed as input?

(I think all other tests modified here have a similar issue)

Reply:
I agree that for lists that have only one item, they are better as variables instead of parameters from parametrize. There are a few functions that have two items in the list on unix, one for sh and one for csh. Should leave those as they are?

Reply:
> There are a few functions that have two items in the list on unix, one for sh and one for csh. Should leave those as they are?

That sounds good to me - thanks!